dove/src/transaction.rs

```diff
@@ -0,0 +1,875 @@
+            .unwrap_or_default()
+            .iter()
+            .map(|signer| self.dove_ctx.dialect.parse_address(signer))
+            .collect::<Result<Vec<_>, Error>>()?;
+        if signers.is_empty() {
+            signers.push(self.dove_ctx.account_address()?);
+        }
+        self.signers = signers;
+        Ok(self)
+    }
+
+    // =============================================================================================
+    /// Create transaction
+    /// Return: Ok(name_transaction, Transaction)
+    pub fn to_transaction(&self) -> Result<(String, Transaction), Error> {
+        let (script, meta) = self.lookup_script()?;
+        let units = self.build_script(script)?;
+
+        let unit = units
+            .into_iter()
+            .find(|unit| {
+                let is_module = match &unit {
+                    CompiledUnit::Module { .. } => false,
+                    CompiledUnit::Script { .. } => true,
+                };
+                is_module && unit.name() == meta.name
+            })
+            .map(|unit| unit.serialize())
+            .map(|mut unit| {
+                self.dove_ctx
+                    .dialect
+                    .adapt_to_target(&mut unit)
+                    .map(|_| unit)
+            })
+            .ok_or_else(|| anyhow!("Script '{}' not found", meta.name))??;
+
+        if meta.type_parameters.len() != self.type_parameters.len() {
+            return Err(anyhow!(
+                "Script '{}' takes {} type parameters, {} passed",
+                meta.name,
+                meta.type_parameters.len(),
+                self.type_parameters.len()
+            ));
+        }
+
+        let (signers, args_count, args) = self.prepare_arguments(&meta.parameters)?;
+
+        if self.args.len() != args_count {
+            return Err(anyhow!(
+                "Script '{}' takes {} parameters, {} passed",
+                meta.name,
+                args_count,
+                self.args.len()
+            ));
+        }
+
+        Ok((
+            meta.name,
+            Transaction::new(signers as u8, unit, args, self.type_parameters.clone())?,
+        ))
+    }
+    /// Find and run the script using the specified parameters
+    pub fn run(&self) -> Result<PipelineExecutionResult, Error> {
+        let (script, _meta) = self.lookup_script()?;
+        let dep_list = self.get_dep_list(script.name().as_ref())?;
+        let executor = Executor::new(self.dove_ctx.dialect.as_ref(), self.signers[0], dep_list);
+        executor.execute_script(script, Some(self.signers.clone()), self.args.clone())
+    }
+    // =============================================================================================
+    fn lookup_script_by_file_name(&self, fname: &str) -> Result<(MoveFile, Meta), Error> {
+        let script_path = self
+            .dove_ctx
+            .path_for(&self.dove_ctx.manifest.layout.scripts_dir);
+        let file_path = if !fname.ends_with("move") {
```

Comment:
this if can be expressed much shortly (given than `set_extensions` updates extension, not appends it):
```rust
let mut file_path = scripts_path.join(file_name);
file_path.set_extension("move");
```